cli/generate_operator_keys.go

```diff
@@ -1,123 +1,125 @@
-			if err != nil {
-				logger.Fatal("Failed to encrypt private key", zap.Error(err))
-			}
-
-			encryptedJSON, err := json.Marshal(encryptedData)
-			if err != nil {
-				logger.Fatal("Failed to marshal encrypted data to JSON", zap.Error(err))
-			}
-
-			err = os.WriteFile("encrypted_private_key.json", encryptedJSON, 0600)
-			if err != nil {
-				logger.Fatal("Failed to write encrypted private key to file", zap.Error(err))
-			}
-
-			logger.Info("private key encrypted and stored in encrypted_private_key.json")
+		if passwordFilePath != "" {
+			encryptAndSavePrivateKey(sk, passwordFilePath, logger)
 		} else {
 			logger.Info("generated public key (base64)", zap.String("pk", base64.StdEncoding.EncodeToString(pk)))
 			logger.Info("generated private key (base64)", zap.String("sk", base64.StdEncoding.EncodeToString(sk)))
 		}
 	},
 }
 
+func readPrivateKeyFromFile(filePath string, logger *zap.Logger) ([]byte, []byte, error) {
+	keyBytes, err := readFile(filePath, logger)
+	if err != nil {
+		return nil, nil, err
+	}
+
+	decodedBytes, err := base64.StdEncoding.DecodeString(string(keyBytes))
+	if err != nil {
+		return nil, nil, err
+	}
+	rsaKey, err := rsaencryption.ConvertPemToPrivateKey(string(decodedBytes))
+	if err != nil {
+		return nil, nil, err
+	}
+
+	skPem := rsaencryption.PrivateKeyToByte(rsaKey)
+
+	operatorPublicKey, err := rsaencryption.ExtractPublicKey(rsaKey)
+	if err != nil {
+		return nil, nil, err
+	}
+	pk, err := base64.StdEncoding.DecodeString(operatorPublicKey)
+	if err != nil {
+		return nil, nil, err
+	}
+	return skPem, pk, nil
+}
+
+func encryptAndSavePrivateKey(sk []byte, passwordFilePath string, logger *zap.Logger) {
+	passwordBytes, err := readFile(passwordFilePath, logger)
+	if err != nil {
+		logger.Fatal("Failed to read password file", zap.Error(err))
+	}
+
+	encryptionPassword := string(passwordBytes)
+	encryptedData, err := keystorev4.New().Encrypt(sk, encryptionPassword)
+	if err != nil {
+		logger.Fatal("Failed to encrypt private key", zap.Error(err))
+	}
+
+	encryptedJSON, err := json.Marshal(encryptedData)
+	if err != nil {
+		logger.Fatal("Failed to marshal encrypted data to JSON", zap.Error(err))
+	}
+	err = writeFile("encrypted_private_key.json", encryptedJSON)
+	if err != nil {
+		logger.Fatal("Failed to write encrypted private key to file", zap.Error(err))
+	}
+
+	logger.Info("private key encrypted and stored in encrypted_private_key.json")
+}
+
+func writeFile(fileName string, data []byte) error {
+	return os.WriteFile(fileName, data, 0600)
+}
+
+func readFile(filePath string, logger *zap.Logger) ([]byte, error) {
+	absPath, err := filepath.Abs(filePath)
+	if err != nil {
+		logger.Fatal(fmt.Sprintf("Failed to read absolute path of %s", filePath), zap.Error(err))
+	}
+	// #nosec G304
+	contentBytes, err := os.ReadFile(absPath)
+	if err != nil {
+		logger.Fatal(fmt.Sprintf("Failed to read file %s", filePath), zap.Error(err))
+	}
+	return contentBytes, err
+}
```

Comment:
this function is weird, you don't need logger if you return errors. return the error and log at the top level instead.